pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/auth/KubernetesSecretsTokenAuthProvider.java

```diff
@@ -97,14 +97,12 @@ public void configureAuthDataStatefulSet(V1StatefulSet statefulSet, Optional<Fun
                         .name(SECRET_NAME)
                         .secret(
                                 new V1SecretVolumeSource()
-                                        .secretName(getSecretName(new String(functionAuthData.get().getData())))
-                                        .defaultMode(256))));
+                                        .secretName(getSecretName(new String(functionAuthData.get().getData()))))));
 
         podSpec.getContainers().forEach(container -> container.setVolumeMounts(Collections.singletonList(
                 new V1VolumeMount()
                         .name(SECRET_NAME)
-                        .mountPath(DEFAULT_SECRET_MOUNT_DIR)
-                        .readOnly(true))));
```

Comment:
Why do we have to drop 'readonly' ?

Reply:
We should keep ready only. It's added back in.